File: Cryptonite/src/Function/Timer.java
package Function;

public class Timer 
{
	private long startTime;
	private long stopTime;
	public Timer() 
	{
		startTime = System.currentTimeMillis();
	}
	
	public void start()
	{
		startTime = System.currentTimeMillis(); 
	}
	
	public void stop()
	{
		stopTime = System.currentTimeMillis();
	}
	
	public long runningTime()
	{
		stop();
		return stopTime - startTime;
	}
	
	public boolean alarm(long time)
	{
		stop();
		
		if((stopTime - startTime) >= time) 
		{
			start();
			return true;
		}
		else
		{
			return false;
		}
	}
}

File: Cryptonite/Cryptonite_Server_Manager/ConsoleApplication15/Timer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication15
{
    class Timer
    {
        private long fTime;
        private long lTime;

        public Timer()
        {
            fTime = DateTime.Now.TimeOfDay.Milliseconds;
            lTime = DateTime.Now.TimeOfDay.Milliseconds;
        }

        public void Start()
        {
            lTime = fTime = DateTime.Now.TimeOfDay.Milliseconds;
        }

        public void Stop()
        {
            lTime = DateTime.Now.TimeOfDay.Milliseconds;
        }

        public Boolean Alarm(long time)
        {
            Stop();

            if(lTime - fTime >= time)
            {
                Start();
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}
